packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-expandable-answer.tsx

```diff
@@ -0,0 +1,116 @@
+import {
+  buildSmartSnippet,
+  SmartSnippet,
+  SmartSnippetState,
+} from '@coveo/headless';
+import {h, Component, State, Prop, Element, Listen} from '@stencil/core';
+import ArrowDown from '../../images/arrow-down.svg';
+import {
+  InitializeBindings,
+  Bindings,
+  BindStateToController,
+} from '../../utils/initialization-utils';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-smart-snippet-expandable-answer',
+  styleUrl: 'atomic-smart-snippet-expandable-answer.pcss',
+  shadow: true,
+})
+export class AtomicSmartSnippetExpandableAnswer {
+  @InitializeBindings() public bindings!: Bindings;
+  public smartSnippet!: SmartSnippet;
+  @BindStateToController('smartSnippet')
+  @State()
+  public smartSnippetState!: SmartSnippetState;
+  public error!: Error;
+  @Element() public host!: HTMLElement;
+
+  /**
+   * The height (in pixels) which, when exceeded by the snippet, displays a "show more" button and truncates the snippet.
+   */
+  @Prop({reflect: true}) minimumSnippetHeightForShowMore = 250;
+  /**
+   * When the answer is partly hidden, how much of its height (in pixels) should be visible.
+   */
+  @Prop({reflect: true}) answerHeightWhenCollapsed = 180;
+
+  @State() showButton = true;
+
+  private validateProps() {
+    if (this.minimumSnippetHeightForShowMore < this.answerHeightWhenCollapsed) {
+      throw 'minimumSnippetHeightForShowMore must be equal or greater than answerHeightWhenCollapsed';
+    }
+  }
+
+  public initialize() {
+    this.validateProps();
+    this.smartSnippet = buildSmartSnippet(this.bindings.engine);
+  }
+
+  public async componentWillRender() {
+    if (this.error) {
+      return;
+    }
+    if (!this.smartSnippetState.answerFound) {
+      throw 'No answer returned by the query';
```

Comment:
Why a throw here ? If someone adds a smart snippet component, there won't be an answer on every query typed by end user.

It's actually going to be the minority of queries that will return an automatic answer.

Should be just a graceful `return` I think.

Reply:
This component is internal and not meant to be used on its own, it should not even be rendered at all when there's no snippet. However, it doesn't do any harm either to make it "usable" on its own, so I'll change this to a return.